Sources/TuistLoader/Loaders/GeneratorModelLoader.swift

```diff
@@ -39,17 +49,35 @@ public class GeneratorModelLoader: GeneratorModelLoading {
         return workspace
     }
 
-    public func loadTuistConfig(at path: AbsolutePath) throws -> TuistCore.TuistConfig {
-        guard let tuistConfigPath = FileHandler.shared.locateDirectoryTraversingParents(from: path, path: Manifest.tuistConfig.fileName) else {
-            return TuistCore.TuistConfig.default
+    public func loadConfig(at path: AbsolutePath) throws -> TuistCore.Config {
+        // If the Config.swift file exists in the root Tuist/ directory, we load it from there
+        if let rootDirectoryPath = rootDirectoryLocator.locate(from: path) {
+            let configPath = rootDirectoryPath.appending(RelativePath("\(Constants.tuistDirectoryName)/\(Manifest.config.fileName)"))
+
+            if FileHandler.shared.exists(configPath) {
+                let manifest = try manifestLoader.loadConfig(at: configPath.parentDirectory)
+                return try TuistCore.Config.from(manifest: manifest)
+            }
+        }
+
+        // We first try to load the deprecated file. If it doesn't exist, we load the new file name.
+        let fileNames = [Manifest.config]
+            .flatMap { [$0.deprecatedFileName, $0.fileName] }
+            .filter { $0 != nil }
+            .map { $0! }
```

Comment:
could this be a `compactMap` ?

---

You could even write an extension so you don't need to specify `$0`

```swift
extension Array {
    
    @inlinable public func compact() rethrows -> [Element] {
        compactMap { $0 }
    }
    
}
```